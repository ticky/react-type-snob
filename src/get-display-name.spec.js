/* global describe, it, expect */
import getDisplayName from './get-display-name';

describe('getDisplayName', () => {
  it('returns expected value for a type with an implicit function name', () => {
    expect(getDisplayName(function ExpectedValue() {})).toMatchSnapshot();
  });

  it('returns expected value for a type with a displayName', () => {
    function component() {}
    component.displayName = 'ExpectedValue';

    expect(getDisplayName(component)).toMatchSnapshot();
  });

  it('returns expected value for a string type with no id or className', () => {
    expect(getDisplayName('ExpectedValue')).toMatchSnapshot();
  });

  it('returns expected value for a string type with an id', () => {
    expect(getDisplayName('div', { id: 'expectedId' })).toMatchSnapshot();
    expect(getDisplayName('div', { id: 'expectedId', className: 'notExpected' })).toMatchSnapshot();
  });

  it('returns expected value for a string type with class names', () => {
    expect(getDisplayName('div', { className: 'expectedClassName anotherExpectedClassName' })).toMatchSnapshot();
  });

  it('falls back when an unidentified type is supplied', () => {
    expect(getDisplayName({})).toMatchSnapshot();
  });
});