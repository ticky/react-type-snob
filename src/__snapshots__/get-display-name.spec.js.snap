exports[`getDisplayName falls back when an unidentified type is supplied 1`] = `"Unknown Element"`;

exports[`getDisplayName returns expected value for a string type with an id 1`] = `"div#expectedId"`;

exports[`getDisplayName returns expected value for a string type with an id 2`] = `"div#expectedId"`;

exports[`getDisplayName returns expected value for a string type with class names 1`] = `"div.expectedClassName.anotherExpectedClassName"`;

exports[`getDisplayName returns expected value for a string type with no id or className 1`] = `"ExpectedValue"`;

exports[`getDisplayName returns expected value for a type with a displayName 1`] = `"ExpectedValue"`;

exports[`getDisplayName returns expected value for a type with an implicit function name 1`] = `"ExpectedValue"`;
